### PR TITLE
enhancement: add description to roles

### PR DESCRIPTION
**Pull request by aberonni**
[AB#30596](https://dev.azure.com/redcrossnl/7d17f2a9-3f18-40f1-8c29-98568fbca7e4/_workitems/edit/30596)

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] I have added tests wherever relevant
- [x] I have made sure that all automated checks pass before requesting a review
- [x] I do not need any deviation from our PR guidelines

### CODE REVIEW

File: services/121-service/test/helpers/utility.helper.ts [hunk 1]
```diff
@@ -113,6 +113,7 @@ export async function removePermissionsFromRole(
   await updatePermissionsOfRole(role.id, {
     label: role.label!,
     permissions: adjustedPermissions as PermissionEnum[],
+    description: role.description!,
```
Comment (jannisvisser) [inline services/121-service/test/helpers/utility.helper.ts:116]: @aberonni why is this needed? the method seems to be only about updating permissions, not about updating description (or label for that matter). Can this be dropped or otherwise renamed?

Comment (aberonni) [reply services/121-service/test/helpers/utility.helper.ts]: That's a good question and a good catch. The properties on `UpdateUserRoleDto` were not marked as optional, which meant you always had to send all properties to be able to update something. I've updated the logic now so that all the properties are optional, and we only update what is sent. That also means that I could remove `label` from this specific call too.

**Comment by aberonni** [comment]
@jannisvisser pinging you because I'm not sure if you have notifications turned on for new review requests and comment responses

**Review by jannisvisser — DISMISSED** [review]
Gave 2 comments that are not related to your changes, but deal with small refactoring related to new guidelines. Up to you to include in this PR or not. For the rest approved.

File: services/121-service/src/user/dto/user-role.dto.ts [hunk 2]
```diff
@@ -8,10 +8,15 @@ export class CreateUserRoleDto {
...
   @IsString()
   public readonly label: string;
 
+  @ApiProperty({ example: 'New user role description' })
+  @IsString()
+  public readonly description: string;
+
   @ApiProperty({
     enum: PermissionEnum,
     example: Object.values(PermissionEnum),
```
[the remark this answers is not in the document]
Comment (aberonni) [reply services/121-service/src/user/dto/user-role.dto.ts:22]: Renaming this would touch at least 40 files so I'm gonna refrain from doing that here to avoid too much of a scope leak

File: services/121-service/src/user/dto/user-role.dto.ts [hunk 3]
```diff
@@ -22,12 +27,19 @@ export class CreateUserRoleDto {
 export class UpdateUserRoleDto {
```
[the remark this answers is not in the document]
Comment (aberonni) [reply services/121-service/src/user/dto/user-role.dto.ts]: Done

**Review by jannisvisser — APPROVED** [review]
